controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -526,22 +560,17 @@ func (r *KubeadmControlPlaneReconciler) getMachines(ctx context.Context, cluster
 	return allMachines, nil
 }
 
-func (r *KubeadmControlPlaneReconciler) getOwnedMachines(ctx context.Context, kcp *controlplanev1.KubeadmControlPlane, clusterName types.NamespacedName) ([]*clusterv1.Machine, error) {
-	allMachines, err := r.getMachines(ctx, clusterName)
-	if err != nil {
-		return nil, err
-	}
-
-	var ownedMachines []*clusterv1.Machine
+func (r *KubeadmControlPlaneReconciler) getOwnedMachines(kcp *controlplanev1.KubeadmControlPlane, allMachines *clusterv1.MachineList) *clusterv1.MachineList {
```

Comment:
I'm curious about why you changed the return from an array to a MachineList?

Reply:
When I made `getOwnedMachines` a filter (and `getMachines` only fetch API resources), I thought it would be easier to use `getOwnedMachines` directly against a MachineList. I didn't see the advantage of copying the MachineList to a slice of Machine pointers.

Reply:
Will fix

Reply:
Mentioned this in https://github.com/kubernetes-sigs/cluster-api/issues/2062